rotkehlchen/externalapis/etherscan.py

```diff
@@ -199,6 +210,13 @@ def _query(
 
             try:
                 result = json_ret.get('result', None)
+                receipt_or_txn_not_found = (
+                    result is None and
```

Comment:
Don't do it like this. There is already a `result is None` check few lines below so you are repeating it. Move the specific action check there as an extra if.